.github/workflows/tests.yml

```diff
@@ -1,43 +1,49 @@
-name: tests
+name: Tests
 
 on:
-  push: null
-  pull_request: null
+  pull_request:
+  push:
+    branches: [master]
 
 jobs:
-  tests:
-    name: tests
-    runs-on: "ubuntu-latest"
+  run:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
+        os: [windows-latest, ubuntu-latest, macos-latest]
+      fail-fast: false
+
     steps:
       - name: cancel previous runs
-        uses: styfle/cancel-workflow-action@0.6.0
+        uses: styfle/cancel-workflow-action@0.10.0
         with:
           access_token: ${{ github.token }}
 
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
-      - uses: conda-incubator/setup-miniconda@v2
+      - name: Setup Micromamba
+        uses: mamba-org/provision-with-micromamba@main
```

Comment:
```suggestion
        uses: mamba-org/provision-with-micromamba@v12
```

Reply:
Indeed this is safer but I have to confess I've been "riding the wave" until today. Let's pin it!

PS: b/c GH releases can be modified to only way to be really safe is to use a specific commit.